src/python/pants/jvm/goals/lockfile.py

```diff
@@ -64,14 +65,18 @@ def wrap_jvm_lockfile_request(request: GenerateJvmLockfile) -> WrappedGenerateLo
 @rule(desc="Generate JVM lockfile", level=LogLevel.DEBUG)
 async def generate_jvm_lockfile(
     request: GenerateJvmLockfile,
+    generate_lockfiles_subsystem: GenerateLockfilesSubsystem,
 ) -> GenerateLockfileResult:
     resolved_lockfile = await Get(CoursierResolvedLockfile, ArtifactRequirements, request.artifacts)
+    regenerate_command = (
+        generate_lockfiles_subsystem.custom_command or f"{bin_name()} generate-lockfiles"
```

Comment:
This whole `bin_name` business should be discontinued given the troubles it has caused with lock file content stability. Its a case of trying to be too helpful causing issues. Clearly for another change though.

Reply:
Yep, agreed